newm-shared/src/main/kotlin/io/newm/shared/serialization/LocalDateSerializer.kt

```diff
@@ -0,0 +1,20 @@
+package io.newm.shared.serialization
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import java.time.LocalDate
+
+// Uses ISO-8601 format
+object LocalDateSerializer : KSerializer<LocalDate> {
```

Comment:
Instead of writing our own serializer, we can probably use the one from `kotlinx-datetime` library. We can add a ticket to the backlog for this though. 

https://github.com/Kotlin/kotlinx-datetime/blob/master/core/common/src/serializers/LocalDateSerializers.kt#L21-L33

Reply:
I looked into the `kotlinx-datetime` library  before. The issue is that they have their own types `kotlinx.datetime.LocalDate` and `kotlinx.datetime.LocalDateTime` instead of supporting the java ones: `java.time.LocalDate` and `java.time.LocalDateTime`. Exposed only works with the later.